h5p_proplist.go

```diff
@@ -40,6 +43,29 @@ func (p *PropList) Close() error {
 	return p.closeWith(h5pclose)
 }
 
+// SetChunk sets the size of the chunks used to store a chunked layout dataset.
+// https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-SetChunk
+func (p *PropList) SetChunk(dim []uint) error {
+	ndims := len(dim)
+	if ndims <= 0 {
+		return h5err(C.herr_t(-1))
+	}
+	c_dim := (*C.hsize_t)(unsafe.Pointer(&dim[0]))
```

Comment:
This is not safe.

Reply:
Do you have a better solution?

Reply:
A for loop.

Reply:
Actually, we have this same pattern in other places.

@sbinet What do you want to do about this? There are many places in the code where there is type punning from `[]uint` to `[]C.hsize_t`. We can't guarantee that this type of unsafe conversion is correct.

I think that if we want to do it, there should be an initial `init` that checks this behaves as expected and panics informatively if not.

Reply:
I already made a change. Please take a look.

Reply:
@kortschak _wrt_ the type prunning: you're probably right.
I've filed: https://github.com/gonum/hdf5/issues/49